src/Hazelcast.Net/NetStandard/StreamExtensions.cs

```diff
@@ -47,41 +48,53 @@ public static async ValueTask<int> ReadAsync(this Stream stream, Memory<byte> me
             if (stream == null) throw new ArgumentNullException(nameof(stream));
             if (memory.Length == 0) throw new ArgumentException("Empty memory.", nameof(memory));
 
-            byte[] bytes = null;
+            // this is based upon what 2.1 does
+
+            if (MemoryMarshal.TryGetArray(memory, out ArraySegment<byte> array))
+            {
+                //return await stream.ReadAsync(array.Array, array.Offset, array.Count, cancellationToken).CAF();
```

Comment:
is this a comment to show below workaround replacement? if so, can you make it no look like comment out code.

Reply:
same - this is what is replaced by the code under it - ok will try to make it more explicit